Cython/Includes/libcpp/memory.pxd

```diff
@@ -85,3 +85,9 @@ cdef extern from "<memory>" namespace "std" nogil:
         shared_ptr[T] lock()
         bool owner_before[Y](const weak_ptr[Y]&)
         bool owner_before[Y](const shared_ptr[Y]&)
+
+    # Smart pointer non-member operations
+    shared_ptr[T] make_shared[T](...) except +
+
+    # Temporaries used for exception handling break generated code
+    unique_ptr[T] make_unique[T](...) # except +
```

Comment:
Yes! Unfortunately, if I leave `except +` uncommented, then the temporaries used for exception handling will cause compilation failures in the generated code, because the C++ compiler wouldn't know that it's safe to perform a move. I've reported this here awhile ago:

https://groups.google.com/forum/#!topic/cython-users/iE_4ZcotEK0

I'm afraid I don't know how to handle the issue properly. It seems that either one needs to somehow suppress the temporaries for this particular function, or add an `std::move()` after the exception handling bloc. I've tried to investigate both, but quickly got lost into the internals...